android/src/main/java/com/swmansion/rnscreens/ScreenFragment.kt

```diff
@@ -155,6 +158,10 @@ open class ScreenFragment : Fragment, ScreenFragmentWrapper {
     override val childScreenContainers: List<ScreenContainer>
         get() = mChildScreenContainers
 
+    fun isTelevision(): Boolean {
+        return context?.getSystemService<UiModeManager>()?.currentModeType == Configuration.UI_MODE_TYPE_TELEVISION
+    }
+
```

Comment:
Is there a particular reason that all changes to the logic are in `ScreenStackFragment` but this method is defined here? 

Just nitpicking here: I would put this method in some kind of util module(`Util.kt`) as such usage does not make much sense for me: `fragmentInstance.isTelevision` (what does it mean that fragment is a television? :D)

My vision:

`fun isPlatformAndroidTV(context: Context?): Boolean`


Also I've noticed that preferred way of resolving the platform [has changed](https://developer.android.com/training/tv/start/hardware.html#runtime-check) since [this  SO answer](https://stackoverflow.com/a/30059511/8635522) :D

Reply:
To be honest I was thinking if I should create a new util class just for one method - that's why I thought about moving it to `ScreenStackFragment`, but you're correct that it should be moved somewhere else 😄

Some things to mention about the technique of detecting a device - I went for the "old" method because for some reason my Android Emulator is still being detected as a non-leanback device 🤔 But that could be my mistake, actually - when I was testing that method I was detecting the device with variable:

```kt
var isTelevision = context?.packageManager?.hasSystemFeature(PackageManager.FEATURE_LEANBACK) == true
```
In this case, `PackageManager#hasSystemFeature` and `UiModeManager#getCurrentModeType` were returning false on Android and Android TV, whereas
```kt
fun isTelevision(): Boolean {
  return context?.getSystemService<UiModeManager>()?.currentModeType == Configuration.UI_MODE_TYPE_TELEVISION
}
```

was returning `true` on Android TV and `false` on Android - so it seems that there is a difference between `variable-statement` and single return in a function. I'll try to test if single return in a function for `PackageManager#hasSystemFeature` will work 👍 

But except this reason I'm also in favor of using UiModeManager because it checks not only the system feature, but the whole device - so we could check in the future if the device is a watch, a car, etc. - with system feature it's possible just check if device has a camera, NFC and other things, related for couple of devices, so for me it could be easier in the future to just add one `case`, instead of changing whole method. What do you think about it @kkafar?

Reply:
```kotlin
var isTelevision = context?.packageManager?.hasSystemFeature(PackageManager.FEATURE_LEANBACK) == true
```

This is evaluated only a single time during variable initialisation. If you want to achieve effect of computed variable you should use:

```kotlin
val isTelevision
    get() = ...
```

If the recommended way of doing things does not work, surely we should go with workaround 😄 But let's make sure it is necessary.

Reply:
@tboba Make sure your test project has `LEANBACK_LAUNCHER` intent filter in AndroidManifest. See https://developer.android.com/training/tv/start/start#tv-activity

Reply:
And maybe also https://developer.android.com/training/tv/start/start#leanback-req

Reply:
@yusufyildirim Thanks for the tips! I've managed to make `hasSystemFeature` work on Android TV without steps you've mentioned, but they may be a good point for investigating why our example apps don't darken the buttons when someone is focusing them 🤔

Reply:
Moved `isTelevision` to DeviceUtils in https://github.com/software-mansion/react-native-screens/commit/f31aee27daa5f09e6a82acdcd8a37e4867b17acd (which is not visible in this PR for some reason at the moment :thinking:)